apps/transport/priv/repo/migrations/20231222145809_improve_epci.exs

```diff
@@ -0,0 +1,37 @@
+defmodule DB.Repo.Migrations.ImproveEpci do
+  use Ecto.Migration
+
+  def change do
+    rename(table(:epci), :code, to: :insee)
+    create(unique_index(:epci, [:insee]))
+
+    alter table(:commune) do
+      add(:epci_insee, references(:epci, column: :insee, type: :string, on_delete: :nilify_all), null: true)
```

Comment:
On peut ajouter une contraine `not null` après avoir rempli la colonne ?

Reply:
Pas pour les communes de la métropole parisienne j'ai l'impression comme tu dis dans la description de la PR

Reply:
Le problème des communes de la métropole parisienne c’est que si on rajoute les EPCI à fiscalité non propre, alors une commune peut appartenir à deux EPCI.

Mais on a bien des communes qui n’appartiennent à aucun EPCI (intercommunalité) : la moitié de la Nouvelle Calédonie, et une poignée d’îles (Ouessant, Ile d’Yeu…). On doit donc laisser la possibilité d’avoir cette colonne sans contenu.

Reply:
Très bien, merci pour les explications 🙏